### PR TITLE
Improve bot resilience

### PR DESCRIPTION
**Pull request by julien-devatom**
### Description of change
- add the new Maker DAI Lender address
- add the possibility to batch the liquidatable user computation. That prevent to have too much RPC failures due to a throughput issue
- fix tests with a newer (more recent) fork block

### Pull-Request Checklist

- [x] Code is up-to-date with the `main` branch
- [x] `yarn test` passes with this change
- [x] This pull request links relevant issues as `Fixes #0000`
- [x] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: src/LiquidationBot.ts [hunk 1]
```diff
@@ -254,24 +257,32 @@ export default class LiquidationBot {
...
-        liquidationsParams.map(async (user) => {
-          if (
-            await this.isProfitable(
-              user.debtMarket.market,
-              user.toLiquidate,
-              user.debtMarket.price
+    this.logger.log(`Found ${users.length} users liquidatable`);
+    // use the batch size to limit the number of users to liquidate
+    const toLiquidate: UserLiquidationParams[] = [];
+    for (let i = 0; i < users.length; i += this.settings.batchSize) {
```
Comment (Rubilmax) [inline src/LiquidationBot.ts:263]: isn't it missing a random delay or a retry strategy?

**Review by Rubilmax — APPROVED** [review]

File: src/LiquidationBot.ts (hunk 1 above)
Comment (julien-devatom) [reply src/LiquidationBot.ts:263]: this is another issue yes, I'm thinking about using rxjs for that

File: src/LiquidationBot.ts [hunk 2]
```diff
@@ -254,24 +257,32 @@ export default class LiquidationBot {
...
-        })
-      )
-    ).filter(Boolean);
+              return user;
+            return null;
+          })
+        )
+      ).filter(Boolean) as UserLiquidationParams[];
+      toLiquidate.push(...batchToLiquidate);
+    }
```
Comment (ghivert) [inline src/LiquidationBot.ts:285]: IMO, Would be better with a `Promise.all` and a `flat` at the end, to highly parallelise the Promise. Or is it done voluntarily ? 🙂

**Review by ghivert — APPROVED** [review]

File: src/LiquidationBot.ts (hunk 2 above)
Comment (julien-devatom) [reply src/LiquidationBot.ts:285]: volontary to prevent RPC throughput errors

**Review by mlgarchery — APPROVED** [review]

**Pull request merged**